src/main/java/mate/academy/hibernate/relations/dao/impl/CountryDaoImpl.java

```diff
@@ -12,11 +15,34 @@ public CountryDaoImpl(SessionFactory sessionFactory) {
 
     @Override
     public Country add(Country country) {
-        return null;
+        Session session = null;
+        Transaction transaction = null;
+        try {
+            session = factory.openSession();
+            transaction = session.beginTransaction();
+            session.save(country);
+            transaction.commit();
+        } catch (Exception e) {
+            if (transaction != null) {
+                transaction.rollback();
+            }
+            throw new DataProcessingException("Can't add country: " + country + "to db", e);
+        } finally {
+            if (session != null) {
+                session.close();
+            }
+        }
+        return country;
     }
 
     @Override
     public Optional<Country> get(Long id) {
-        return null;
+        Country country = null;
+        try (Session session = factory.openSession()) {
+            country = session.get(Country.class, id);
+        } catch (Exception e) {
+            throw new DataProcessingException("Can't get country: " + id + "from db", e);
+        }
+        return Optional.ofNullable(country);
```

Comment:
can you avoid using redundant 'country' variable? Fix the same issue in all places